src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/EditEndDateRequestCommandHandlerTests.cs

```diff
@@ -0,0 +1,128 @@
+        public ProviderCommitmentsDbContext Db { get; set; }
+
+        public UnitOfWorkContext UnitOfWorkContext { get; set; }
+
+        public Party Party { get; set; }
+
+        public long ApprenticeshipId { get; set; }
+
+        public EditEndDateRequestCommandHandlerTestsFixture()
+        {
+            Party = Party.Employer;
+            UnitOfWorkContext = new UnitOfWorkContext();
+            Db = new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+                                                 .UseInMemoryDatabase(Guid.NewGuid().ToString())
+                                                 .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                                                 .Options);
+            var fixture = new Fixture();
+            fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+
+            var Cohort = new CommitmentsV2.Models.Cohort()
+               .Set(c => c.Id, 111)
+               .Set(c => c.EmployerAccountId, 222)
+               .Set(c => c.ProviderId, 333);
+            var Apprenticeship = fixture.Build<CommitmentsV2.Models.Apprenticeship>()
+             .With(s => s.Cohort, Cohort)
+             .With(s => s.PaymentStatus, PaymentStatus.Completed)
+             .With(s => s.EndDate, DateTime.UtcNow)
+             .With(s => s.CompletionDate, DateTime.UtcNow.AddDays(10))
+             .With(s => s.StartDate, DateTime.UtcNow.AddDays(-10))
+             .Without(s => s.DataLockStatus)
+             .Without(s => s.EpaOrg)
+             .Without(s => s.ApprenticeshipUpdate)
+             .Without(s => s.Continuation)
+             .Without(s => s.PreviousApprenticeship)
+             .Create();
+
+            Db.Apprenticeships.Add(Apprenticeship);
+
+            Db.SaveChanges();
+
+            ApprenticeshipId = Apprenticeship.Id;
+
+            var authenticationService = new Mock<IAuthenticationService>();
+            authenticationService.Setup(x => x.GetUserParty()).Returns(() => Party);
+
+            var lazyProviderDbContext = new Lazy<ProviderCommitmentsDbContext>(() => Db);
+
+            var newEndDate = Apprenticeship.EndDate.Value.AddDays(1);
+
+            Command = new EditEndDateRequestCommand
+            {
+                ApprenticeshipId = Apprenticeship.Id,
+                EndDate = newEndDate,
+                UserInfo = new UserInfo()
+            };
+
+            Handler = new EditEndDateRequestCommandHandler(lazyProviderDbContext,
+                Mock.Of<ICurrentDateTime>(),
+                authenticationService.Object,
+                Mock.Of<ILogger<EditEndDateRequestCommandHandler>>());
+        }
+
+        public async Task Handle()
+        {
+            await Handler.Handle(Command, CancellationToken.None);
+            await Db.SaveChangesAsync();
+        }
+
+        internal void SetParty(Party party)
```

Comment:
no need for SetParty() the property has a public setter

Reply:
done.